### PR TITLE
Compile resources in subdirectories correctly.

### PR DESCRIPTION
**Pull request by andr-dots**
Those changes allows this project to be in a subdirectory (with add_subdirectory) of main project that uses it.

### CODE REVIEW

**Comment by Makman2** [comment]
@9fcc hey :D
wow cool someone has seen my GCR_CMake extensions I've used only privately for a single project, that is _ugh_ currently not in active development :3

Thanks for submitting the PR! :D Changes look good, some minor stuff I would ask you to change (via `git commit --amend`, see inline comments), then I would merge that directly and make a new release :)

File: README.md [hunk 1]
```diff
@@ -127,7 +127,7 @@ compile_gresources(<output>
 - **SOURCE_DIR** ***resource_directory***
```
Comment (Makman2) [inline README.md:127]: commit message:
could you remove the trailing period? :)
( I know this is a bit nit-picky, but I'm currently working on https://github.com/coala/coala which is all about code quality, involving good commit messages :3)

File: macros/CompileGResources.cmake [hunk 2]
```diff
@@ -47,7 +47,7 @@ function(COMPILE_GRESOURCES output xml_out)
     # RESOURCES The list of resource files. Whether absolute or relative path is
     #           equal, absolute paths are stripped down to relative ones. If the
     #           absolute path is not inside the given base directory SOURCE_DIR
-    #           or CMAKE_SOURCE_DIR (if SOURCE_DIR is not overriden), this
+    #           or CMAKE_CURRENT_SOURCE_DIR (if SOURCE_DIR is not overriden),
```
Comment (Makman2) [inline macros/CompileGResources.cmake:50]: could you put `this` here?

File: macros/GenerateGXML.cmake [hunk 3]
```diff
@@ -26,7 +26,7 @@ function(GENERATE_GXML xml_path)
     # RESOURCES The list of resource files. Whether absolute or relative path is
     #           equal, absolute paths are stripped down to relative ones. If the
     #           absolute path is not inside the given base directory SOURCE_DIR
-    #           or CMAKE_SOURCE_DIR (if SOURCE_DIR is not overriden), this
+    #           or CMAKE_CURRENT_SOURCE_DIR (if SOURCE_DIR is not overriden),
```
Comment (Makman2) [inline macros/GenerateGXML.cmake:29]: here too :)

**Review by Makman2 — CHANGES_REQUESTED** [review]

**Comment by Makman2** [comment]
If you don't want to fixup the stuff I've mentioned, I can take over for you, making a new PR and modifying your commit (don't fear, the author information is kept :D)

File: README.md (hunk 1 above)
Comment (andr-dots) [reply README.md:127]: Done.

**Comment by Makman2** [comment]
@9fcc many thanks!! :D I really appreciate this PR :) Haven't imagined someone would stumble upon these macros :)

**Comment by andr-dots** [comment]
Well, we used those macros for a long time in our project and they saved some my time. It goes as a submodule in git in a subdirectory so original version wasn't perfect. At the moment we decided to push changes back to original developer. If GTK+ someday would become popular those macros would be very useful. Thanks.

**Pull request merged**